### ISSUE TITLE
How to enforce a hierarchy for parent/child relationships?

### ISSUE THREAD

**Issue by mintyfusion** [casbin/casbin#315]
We have a structure like this:

Grand Parent
      Parent
          Children
                Grand Children
                        So on....

I have two questions:

1) If user has permission on Parent, how do we find out the grand parent to show as View?
2) If the user navigates to Grand Children directly for that specific parent, how do we perform that check, and allow access?

Thanks,
Harsimrat

**Comment by jkour** [comment]
If the user has Parent View permission, then they must have Grand Parent view permissions to do what you say

**Comment by mintyfusion** [comment]
That is correct, the way our Web/App is setup is that you need View Permissions to Grand Parent to navigate to Parent.

In addition, if user navigates to Grand Children, we would like to know that they have Permission to Parent and then we can trim specific actions based on the role.

**Comment by jkour** [comment]
In Casbin4D there is a method that returns the permissions for a user. For example, you would get permissionsForUser('Grand Children') and you would get all of them.

I am sure @hsluoyz can help you with the Go implementation.

Also, I am not clear with your first question: do you want to find our who the Grant Parent is? If so, I am not sure it is part of the casbin model.

**Comment by mintyfusion** [comment]
We will be using .NET casbin, and I will investigate if permissionForUser is available there or not. 

Yes, we would like to know who the Grant Parent is to show that information in View only mode.

**Comment by user4517** [comment]
Anyone using this to provide ABAC support using the XACML framework?

**Comment by hsluoyz** [comment]
This issue transferred to Casbin.NET
